python/ucxx/_lib/libucxx.pyx

```diff
@@ -40,15 +41,45 @@ include "tag.pyx"
 logger = logging.getLogger("ucx")
 
 
-np.import_array()
-
+cdef class HostBufferAdapter:
```

Comment:
As we already have [`Array`]( https://github.com/rapidsai/ucxx/blob/bdd3ff95a6646f642543095bccee3c7b0db3a985/python/ucxx/_lib/arr.pyx#L73 ), could we just use that? Or if `Array` is missing something, could we add whatever we need to it?

If possible, would like to move towards one object that does this job for us consistently

Reply:
I don't think that makes sense here. As we discussed in the meeting, Array and objects like it are designed to be consumers of various protocols, but they shouldn't support raw pointers. Any object that we want to be consumable by Array needs to export one of the supported protocols. In this case, we're trying to construct a numpy array from a C++ object, the HostBuffer. Unless we want to implement the buffer protocol on the HostBuffer object directly (that requires changing ucxx's C++ code, but is an option), the easiest thing we can do is produce a wrapper around it that does export the protocol. 

The more appropriate usage of Array here would be to use it in place of the numpy array that we're returning, since that that point we're using numpy as a consumer of the buffer protocol and the Array should be able to do that too. That would be an API-breaking change though, and not one that buys us anything.

Reply:
> Unless we want to implement the buffer protocol on the HostBuffer object directly (that requires changing ucxx's C++ code, but is an option)

Yes this is what I'm suggesting

Reply:
But then how would you construct the Array from the HostBuffer?

Reply:
If I'm understanding his statements correctly, @seberg and I are basically in agreement here. @jakirkham could you elaborate on how you envisioned this working and whether the above discussion clears up any confusion?

Reply:
Right, I think what makes this object distinct and useful is that it owns its data and `Array` does not.  That seems like a useful split.
That probably means it won't be a helper for exporting the buffer protocol elsewhere, but that is probably good as the code here is simple and doesn't have to deal with complexities like shapes, strides, dtypes, ….